municipal_finance/admin.py

```diff
@@ -62,26 +65,95 @@ def save_model(self, request, obj, form, change):
         )
 
 
-@admin.register(MunicipalityStaffContactsUpload)
-class MunicipalityStaffContactsUploadAdmin(admin.ModelAdmin):
+@admin.register(MunicipalStaffContactsUpdate)
+class MunicipalStaffContactsUpdateAdmin(admin.ModelAdmin):
     list_display = ('datetime',)
     readonly_fields = ('user',)
 
     def get_exclude(self, request, obj=None):
         if obj is None:
             return ('user',)
         else:
-            return super(MunicipalityStaffContactsUploadAdmin, self).get_exclude(request, obj)
+            return super(MunicipalStaffContactsUpdateAdmin, self).get_exclude(request, obj)
 
     def save_model(self, request, obj, form, change):
         # Set the user to the current user
         obj.user = request.user
         # Process default save behavior
-        super(MunicipalityStaffContactsUploadAdmin,
-              self).save_model(request, obj, form, change)
+        super(MunicipalStaffContactsUpdateAdmin, self).save_model(
+            request, obj, form, change
+        )
         # Queue task
         async_task(
             'municipal_finance.update.update_municipal_staff_contacts',
             obj,
-            task_name='Municipal staff contacts upload',
+            task_name='Municipal staff contacts update',
+        )
+
+
+class BaseUpdateAdmin(admin.ModelAdmin):
+    list_display = ('user', 'datetime', 'deleted', 'inserted',)
+    readonly_fields = ('user', 'deleted', 'inserted',)
+
+    def get_exclude(self, request, obj=None):
+        if obj is None:
+            return ('user',)
+        else:
+            return super(BaseUpdateAdmin, self).get_exclude(request, obj)
+
+
+@admin.register(IncomeExpenditureV2Update)
+class IncomeExpenditureV2UpdateAdmin(BaseUpdateAdmin):
+
+    def save_model(self, request, obj, form, change):
```

Comment:
we might want to check if it's new or an edit later using `change` I think, since you probably don't want changes that come about who knows how to accidentally re-import and perhaps overwrite later files.

but that can be added to the backlog to do later.

Reply:
@jbothma I've added a check. I'll add a task to the backlog to reflect this behaviour on the interface.